packages/boot/test/acceptance/controller.booter.app.acceptance.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright IBM Corp. 2013,2018. All Rights Reserved.
+// Node module: @loopback/boot
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Client, createClientForHandler} from '@loopback/testlab';
+import {ControllerBooterApp} from '../fixtures/booterApp/application';
+import {ApplicationConfig} from '@loopback/core';
+import {RestServer} from '@loopback/rest';
+import {ControllerBooter} from '../../index';
+// @ts-ignore
+import {getCompilationTarget} from '@loopback/build/bin/utils';
+
+describe('controller booter acceptance tests', () => {
+  let app: ControllerBooterApp;
+  let appConfig: ApplicationConfig;
+  let projectRoot: string;
+
+  beforeEach(getProjectRoot);
+  beforeEach(getAppConfig);
+  beforeEach(getApp);
+
+  afterEach(stopApp);
+
+  it('binds controllers using ControllerDefaults and REST endpoints work', async () => {
+    app.booter(ControllerBooter);
+    await app.boot();
+    await app.start();
+
+    const server: RestServer = await app.getServer(RestServer);
+    const client: Client = createClientForHandler(server.handleHttp);
+
+    // Default Controllers = /controllers with .controller.js ending (nested = true);
+    await client.get('/nested').expect(200, 'NesterController.nester()');
+    await client.get('/').expect(200, 'HelloController.hello()');
+    await client.get('/one').expect(200, 'ControllerOne.one()');
+    await client.get('/two').expect(200, 'ControllerOne.two()');
+  });
+
+  function getProjectRoot() {
+    let dist = 'dist';
+    if (getCompilationTarget() === 'es2015') dist = 'dist6';
+    projectRoot =
+      process.cwd().indexOf('packages') > -1
+        ? `${dist}/test/fixtures/booterApp`
+        : `packages/boot/${dist}/test/fixtures/booterApp`;
```

Comment:
I consider it as a code smell that this code needs to know about what is our compilation target and what is the layout of `dist` directories used by the compiler.

I am proposing to use a simpler solution based on `__dirname` or `require.resolve`, for example:

```ts
projectRoot = path.resolve(__dirname, '..', 'fixtures', 'booterApp');
// or
projectRoot = path.dirname(require.resolve('../fixtures/booterApp/application.ts'));
```

(This requires that fixtures are copied or transpiled as part of our build. The first alternative can be achieved by adding `--allowJs` (possibly together with `--checkJs`) to `tsc` [compiler options](https://www.typescriptlang.org/docs/handbook/compiler-options.html). The second alternative requires us to change the dist files from `.js` to `.ts`. I personally prefer the second option, as it's closer to how our users are going to use the bootstrapper. IIUC, this part is already solved.)